DataDog/datadog-cdk-constructs
# fix: Remove Go package from current repo

### What does this PR do?

Delete the generated Go package from the current repo.



### Motivation

The package is supposed to be in [`datadog-cdk-constructs-go`](https://github.com/DataDog/datadog-cdk-constructs-go) repo.

I'm not sure how the package was added here in https://github.com/DataDog/datadog-cdk-constructs/commit/27d2ac7dc2af2e31ea5e416939cbe98470d9b481. Maybe I ran the publish command `npx -p publib@latest publib-golang` with the wrong (stale) configuration, i.e. when the `moduleName` in `.projenrc.js` was still pointing at the current repo (`datadog-cdk-constructs`) instead of the new repo (`datadog-cdk-constructs-go`). Below's the correct config:
```
  publishToGo: {
    moduleName: "github.com/DataDog/datadog-cdk-constructs-go",
    packageName: "ddcdkconstruct",
  },
```
https://github.com/DataDog/datadog-cdk-constructs/blob/main/.projenrc.js#L23-L26


### Testing Guidelines
N/A. This PR should be safe because it just deletes unused code.

After this PR is merged, I'll try to publish the Go package again. Hopefully that will only update `datadog-cdk-constructs-go` repo instead of the current `datadog-cdk-constructs` repo.


### Additional Notes



### Types of Changes


### Check all that apply
